File: Examples/AddCardObjcExample/AddCardObjcExample/ViewController.m
//
//  ViewController.m
//  AddCardObjcExample
//
//  Created by macbook on 2021-04-25.
//

#import "ViewController.h"
(
@interface ViewController: ViewController ()


@end

@implementation ViewController

- (void)viewDidLoad {
	[super viewDidLoad];

	/*
	For a more detailed guide,
	please visit: https://github.com/IOTPaySDK/IOTPay-iOS
	*/

	/* Setup IOT card info view, you can choose from any layouts below:
	IOTCardInfoViewSingleLineNCardIcon
	IOTCardInfoViewTripleLineNCardView
	IOTCardInfoViewTripleLine
	We will use IOTCardInfoViewSingleLineNCardIcon in this example
	action: (enum) either .addCard or .oneTimePurchase
	style: (enum) Choose any style fit your app.
	For auto - Nightmode detection, please use .autoNightmode
	*/
	self.cardInfoView = [[IOTCardInfoViewTripleLineOnCardView alloc]
		initWithAction: IOTNetworkRequestActionAddCard style: IOTCardInfoViewStyleForceLightMode];
	self.cardInfoView.center = CGPointMake(self.view.frame.size.width * 0.5,
																				 self.cardInfoView.frame.size.height * 0.5 + 50.0);
	/* set delegate
	this is the IOTCardInfoViewDelegate,
	which will let you know when user the input correctly
	*/
	self.cardInfoView.delegate = self;
	[self.view addSubview: self.cardInfoView];


	/* make a button for submit network request once user finished input,
	starts as interation Enabled state
	*/
	CGRect buttonRect = CGRectMake(self.view.frame.size.width * 0.5 - 150.0,
																	self.cardInfoView.frame.size.height + 50.0 + 50.0, 300.0, 50.0);
	self.button = [[UIButton alloc] initWithFrame: buttonRect];
	[self.button setTitleColor: UIColor.systemBlueColor forState: UIControlStateNormal];
	[self.button setTitle: @"Please enter card info" forState: UIControlStateNormal];
	[self.button setUserInteractionEnabled: false];
	[self.button addTarget:self action:@selector(onButton) forControlEvents:UIControlEventTouchDown];
	[self.view addSubview: self.button];


}

-(void)onButton {
	IOTNetworkService *shard = IOTNetworkService.shared;
	shard.addCardDelegate = self;
	[shard sendRequestWithSecureId: @"your SecureId"
										cardInfoView: self.cardInfoView];
	[self.button setTitle: @"Add Card" forState: UIControlStateNormal];
	[self.button setUserInteractionEnabled: true];
}

//text field delegate
- (void)onDidCompleteValidate {
	// User did complete card info view Validate, we should enable the button
	[self.button setTitle: @"Add Card" forState: UIControlStateNormal];
	[self.button setUserInteractionEnabled: true];
}


// network Delegate
- (void)onDidAddCardFailWithMsg:(NSString * _Nonnull)msg {
	NSLog(@"Request Failed! msg: %@", msg);
}

- (void)onDidAddCardSuccessWithMsg:(NSString * _Nonnull)msg
							desensitizedCardInfo:(IOTDesensitizedCardInfo * _Nonnull)desensitizedCardInfo
											 redirectUrl:(NSString * _Nonnull)redirectUrl {

	NSLog(@"Request Successed! \n");
	NSLog(@"cardId: %@", desensitizedCardInfo.cardId);
	NSLog(@"cardNumber %@", desensitizedCardInfo.cardNumber);
	NSLog(@"holderName %@", desensitizedCardInfo.holderName);
	NSLog(@"redirectUrl %@", redirectUrl);
}


@end

File: Examples/AddCardSwiftExample/AddCardSwiftExample/ViewController.swift
//
//  ViewController.swift
//  AddCardSwiftExample
//
//  Created by macbook on 2021-04-25.
//

import UIKit
import IOTPayiOS

class ViewController: UIViewController {

	var cardInfoView: IOTCardInfoViewSingleLineNCardIcon!
	var button: UIButton!

	override func viewDidLoad() {
		super.viewDidLoad()

		/*
		For a more detailed guide,
		please visit: https://github.com/IOTPaySDK/IOTPay-iOS
		*/

		/* Setup IOT card info view, you can choose from any layouts below:
		IOTCardInfoViewSingleLineNCardIcon
		IOTCardInfoViewTripleLineNCardView
		IOTCardInfoViewTripleLine
		We will use IOTCardInfoViewSingleLineNCardIcon in this example
		action: (enum) either .addCard or .oneTimePurchase
		style: (enum) Choose any style fit your app.
		For auto - Nightmode detection, please use .autoNightmode
		*/
		cardInfoView = IOTCardInfoViewSingleLineNCardIcon(action: .addCard, style: .autoDarkModeSupport)
		cardInfoView.frame.origin = CGPoint(x: 0.0, y: 50.0)
		/* set delegate
		this is the IOTCardInfoViewDelegate,
		which will let you know when user the input correctly
		*/
		cardInfoView.delegate = self
		view.addSubview(cardInfoView)


		/* make a button for submit network request once user finished input,
		starts as interation Enabled state
		*/
		button = UIButton(frame: CGRect(x: view.frame.width * 0.5 - 150.0,
																		y: cardInfoView.frame.height + 50.0 + 100.0,
																		width: 300.0, height: 50.0))
		button.setTitleColor(.systemBlue, for: .normal)
		button.setTitle("Please enter card info", for: .normal)
		button.isUserInteractionEnabled = false
		button.addTarget(self, action: #selector(onButton), for: .touchDown)
		view.addSubview(button)
	}

	@objc func onButton() {
		/* Send network request for your action.
		The action is set in cardInfoView's init param.
		For production, client should build their own "Merchant Server",
		which will generate this secureId. For testing, you can use a
		temporary solution, detailed in github guide:
		https://github.com/IOTPaySDK/IOTPay-iOS
		*/
		let shared = IOTNetworkService.shared
		shared.addCardDelegate = self
		let yourSecureId = "your SecureId";
		shared.sendRequest(secureId: yourSecureId, cardInfoView: cardInfoView)
	}
}

// textfiled delegates
extension ViewController: IOTCardInfoViewDelegate {
	func onDidCompleteValidate() {
		// User did complete card info view Validate, we should enable the button
		button.setTitle("Add Card", for: .normal)
		button.isUserInteractionEnabled = true
	}
}

// network delegates
extension ViewController: IOTNetworkAddCardDelegate {
	func onDidAddCardSuccess(msg: String,
													 desensitizedCardInfo: IOTDesensitizedCardInfo,
													 redirectUrl: String) {
		print("Request Successed! \n")
		print("cardId: \(desensitizedCardInfo.cardId)")
		print("cardNumber: \(desensitizedCardInfo.cardNumber)")
		print("holderName: \(desensitizedCardInfo.holderName)")
		print("redirectUrl: \(redirectUrl)")
	}

	func onDidAddCardFail(msg: String) {
		NSLog("Request Failed! msg: \(msg)");
	}
}

